spinalcordtoolbox/reports/slice.py

```diff
@@ -38,26 +38,34 @@ class Slice(object):
 
     def __init__(self, images, p_resample=0.6):
         """
-        :param images: list of 3D volumes to be separated into slices.
+        :param images: list of 3D or 4D volumes to be separated into slices.
         """
         logger.info('Resample images to {}x{} mm'.format(p_resample, p_resample))
-        self._images = list()
+        self._images = list()  # 3d volumes
+        self._4d_images = list()  # 4d volumes
+        self._image_seg = None  # for cropping
```

Comment:
I have some possible suggestions that could simplify the changes to `Slice`:

* Instead of `self._image_seg`, call it `self._image_ref`
* Set `self._image_ref` to  `self._images[0]` inside `__init__`, then replace any calls to `self._images[0]`.
* Because you no longer need to access  `_images[0]` in `mosaics_through_time`, I think you no longer need to keep both `_images` and `_images_4d`, which lets you use the same variable for both 3D and 4D cases.

Reply:
I was stuck for a while on this issue when writing 'mosaics_through_time'! I created '_images_4d' mainly to be able to use `mosaic` from without too much changes. My initial version was just like you suggested with 'self._images' being used for both 3D and 4D images, but the code was quite messy with a lot of if/else blocks!

Reply:
Gotcha! That makes sense... no worries, then! :slightly_smiling_face: